fractions/microprofile/microprofile-fault-tolerance/src/main/java/org/wildfly/swarm/microprofile/faulttolerance/deployment/DefaultCommand.java

```diff
@@ -30,26 +34,56 @@
      * @param setter
      * @param ctx
      * @param fallback
+     * @param operation
      */
-    protected DefaultCommand(Setter setter, ExecutionContextWithInvocationContext ctx, Supplier<Object> fallback) {
+    protected DefaultCommand(Setter setter, ExecutionContextWithInvocationContext ctx, Supplier<Object> fallback, FaultToleranceOperation operation) {
         super(setter);
         this.ctx = ctx;
         this.fallback = fallback;
+        this.failure = null;
+        this.operation = operation;
     }
 
     @Override
     protected Object run() throws Exception {
-        return ctx.proceed();
+        try {
+            return ctx.proceed();
+        } catch (Exception e) {
```

Comment:
`failOn` defaults to `Throwable`, so maybe you need to catch that here?

Reply:
Good point, although I don't expect people using `Error`s here ;-)